trieste/models/gpflow/interface.py

```diff
@@ -78,10 +84,18 @@ def model(self) -> GPModel:
         """The underlying GPflow model."""
 
     def predict(self, query_points: TensorType) -> tuple[TensorType, TensorType]:
-        return (self._posterior or self.model).predict_f(query_points)
+        mean, cov = (self._posterior or self.model).predict_f(query_points)
+        # posterior predict has been observed to return negative values; stop it
+        if self._posterior is not None:
+            cov = tf.clip_by_value(cov, 1e-12, cov.dtype.max)
```

Comment:
Is someone looking into this? Looks a bit dodge but fine as a fix if temporary .

Reply:
I'm raising a gpflow issue, but I think this will require a researcher to investigate. Are you volunteering? :-)

Also, next time you should probably wait until the tests pass before ticking as you missed an oopsie: I was making all the covariances in predict_joint positive too! I've fixed it so it only updates the variances. (Alternatively I could skip changing skip_joint as it doesn't currently make any tests fail, but it feels bad to knowingly return incorrect values.)

Reply:
PS I'll add the gpflow issue number to the comments